services/fileuploader/fileuploader.go

```diff
@@ -12,17 +15,50 @@ type FileUploader interface {
 
 // SettingsT sets configuration for FileUploader
 type SettingsT struct {
-	Provider       string
-	AmazonS3Bucket string
+	Provider string
+	Bucket   string
 }
 
 // NewFileUploader returns FileFileUploader backed by configured privider
 func NewFileUploader(settings *SettingsT) (FileUploader, error) {
 	switch settings.Provider {
 	case "s3":
 		return &S3Uploader{
-			bucket: settings.AmazonS3Bucket,
+			bucket: settings.Bucket,
+		}, nil
+	case "minio":
+		return &MinIOUploader{
+			bucket:          settings.Bucket,
+			endpoint:        config.GetEnv("MINIO_ENDPOINT", ""),
+			accessKeyID:     config.GetEnv("MINIO_ACCESS_KEY_ID", ""),
+			secretAccessKey: config.GetEnv("MINIO_SECRET_ACCESS_KEY", ""),
+			useSSL:          config.GetBool("MINIO_USE_SSL", false),
 		}, nil
 	}
-	return nil, errors.New("No provider configured for FileUploader")
+	return nil, errors.New("No provider configured for FileUploader: " + settings.Provider)
+}
+
+func CreateUploaderForDestination(destination backendconfig.DestinationT) (FileUploader, error) {
+
+	switch destination.DestinationDefinition.Name {
+	case "s3":
+		return &S3Uploader{
+			bucket: destination.Config.(map[string]interface{})["bucketName"].(string),
+		}, nil
+	case "minio":
+
+		return &MinIOUploader{
+			bucket:          destination.Config.(map[string]interface{})["bucketName"].(string),
+			endpoint:        destination.Config.(map[string]interface{})["MINIO_ENDPOINT"].(string),
+			accessKeyID:     destination.Config.(map[string]interface{})["MINIO_ACCESS_KEY_ID"].(string),
+			secretAccessKey: destination.Config.(map[string]interface{})["MINIO_SECRET_ACCESS_KEY"].(string),
+			useSSL:          destination.Config.(map[string]interface{})["MINIO_USE_SSL"].(bool),
```

Comment:
@SumanthPuram do you think it would be better to rename these parameters to lowercase unprefixed to not confuse them with environment variables? I would suggest:
```
endpoint
accessKey
secretKey
useSSL
```

Reply:
Yes @obukhov We should mark them camelCase. Can you please make the change? I will test this and merge the PR.